src/DynamoCoreWpf/Views/Search/LibraryView.xaml.cs

```diff
@@ -285,5 +286,67 @@ private void OnAddButtonPreviewMouseDown(object sender, MouseButtonEventArgs e)
 
             e.Handled = true;
         }
+
+        /// <summary>
+        /// Position of mouse, when user clicks on button.
+        /// </summary>
+        private Point startPosition;
+
+        /// <summary>
+        /// Indicates whether item is dragging or not, so that there won't be more than one DoDragDrop event.
+        /// </summary>
+        private bool IsDragging;
+
+        private void OnExpanderButtonPreviewMouseMove(object sender, MouseEventArgs e)
+        {
+            if (e.LeftButton != MouseButtonState.Pressed)
+                return;
+
+            Point currentPosition = e.GetPosition(null);
+
+            // If item was dragged enough, then fire DoDragDrop. 
+            // Otherwise it means user click on item and there is no need to fire DoDragDrop.
+            if ((System.Math.Abs(currentPosition.X - startPosition.X) > SystemParameters.MinimumHorizontalDragDistance ||
+                System.Math.Abs(currentPosition.Y - startPosition.Y) > SystemParameters.MinimumVerticalDragDistance) &&
+                !IsDragging)
+            {
+                StartDrag(e);
+            }
+
+        }
+
+        private void StartDrag(MouseEventArgs e)
+        {
+            IsDragging = true;
+            var senderButton = e.OriginalSource as FrameworkElement;
+
+            var searchElementVM = senderButton.DataContext as NodeSearchElementViewModel;
+            if (searchElementVM == null)
+            {
+                IsDragging = false;
+                return;
+            }
+
+            DragDrop.DoDragDrop(senderButton, new DragDropNodeSearchElementInfo(searchElementVM.Model), DragDropEffects.Copy);
```

Comment:
Should we only set `IsDragging = true` right before `DragDrop.DoDragDrop`? Or even better, can we do it only inside `DragDrop.DoDragDrop` method so it becomes self-contained?